test/box-luatest/gh_7972_func_def_err_conditions_test.lua

```diff
@@ -0,0 +1,194 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function(cg)
+    cg.server = server:new({alias = 'dflt'})
+    cg.server:start()
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+-- Checks uncovered function definition error conditions (see also coverage
+-- report https://coveralls.io/builds/56509110/source?filename=src%2Fbox%2Falter.cc).
+g.test_func_def_err_conditions = function(cg)
+    cg.server:exec(function()
+        local t = require('luatest')
+
+        local function setmap(table)
+            return setmetatable(table, { __serialize = 'map' })
+        end
+
+        t.assert_error_msg_equals("Tuple field 3 (name) type does not " ..
+                                  "match one required by operation: " ..
+                                  "expected string, got nil", function()
+            box.space._func:auto_increment{box.session.euid(), nil, 0, 'lua',
+                                           '', 'function', {}, 'any', 'none',
+                                           'none', false, false, true, {},
+                                           setmap{}, '',
+                                           os.date("%Y-%m-%d %H:%M:%S"),
+                                           os.date("%Y-%m-%d %H:%M:%S")}
```

Comment:
Nit: You could cache the date and then reuse it to save a few lines:

```Lua
local date = os.date("%Y-%m-%d %H:%M:%S")
```

Reply:
Fixed.